src/ccnl-core/include/ccnl-prefix.h

```diff
@@ -116,11 +116,14 @@ ccnl_prefix_cmp(struct ccnl_prefix_s *pfx, unsigned char *md,
  *
  * @param[in] prefix        Prefix to be compared (shorter of Longest Prefix matching)
  * @param[in] minsuffix     ?
- * @param[in] maxsuffix     ?
+ * @param[in] maxsuffix     ? 
  * @param[in] c             Content to test the prefix against
  *
- * @return      0 if no match
- * @return      1 if full match
+ * @return      -1 if no match at all (all modes) or exact match failed
```

Comment:
The doc is as proposed not really true:
```
    int32_t cmp = ccnl_prefix_cmp(p, md, prefix, CMP_MATCH);
    return cmp > 0 && (uint32_t) cmp == prefix->compcnt;
```

due to this last line, it is never possible to return `-1`.

Reply:
You're right. I change that in #367 ([here](https://github.com/cn-uofbasel/ccn-lite/pull/367/files#diff-4f941be7fa6d08836de5a14bfb2e17b1R474)). So how about we live with this little inconsistency (which was already there **before** merging #357) and focus on #367? Furthermore, keeping the revert commit as is seems like a clean solution for the git history.

Reply:
Okay, would've been clearer if both commits were in the same PR as they seem to belong together.